siac/cmds.go

```diff
@@ -47,20 +48,64 @@ func updatecmd(cmd *cobra.Command, args []string) {
 		cmd.Usage()
 		return
 	}
-	fmt.Println(getResponse("/update", &url.Values{
-		"action": {args[0]},
-	}))
+	switch args[0] {
+	case "check":
+		if m, err := getResponse("/update/check", nil); err == nil {
+			if m["Available"].(bool) {
+				fmt.Printf("Update %s is available!\nRun 'siac update %s' to install it.\n", m["Version"], m["Version"])
+			} else {
+				fmt.Println("Up to date!")
+			}
+		} else {
+			fmt.Println("Update check failed:", err)
+		}
```

Comment:
You say these things like they're known facts, but until we actually *have* more developers it's all just speculation. Not to mention premature optimization.

Consistency is nice, and I don't think your logic is totally wrong here. But ultimately I don't think it's a terrible burden on other developers. IMHO, it's much more important that code is:
- modular (i.e. non-monolithic; split into smaller sub-functions with clear inputs+outputs)
- well-documented
- well-tested
...than whether or not it follows some particular convention.